wandb/sdk/launch/builder/build.py

```diff
@@ -333,27 +335,56 @@ def get_requirements_section(launch_project: LaunchProject, builder_type: str) -
         buildx_installed = False
     if launch_project.deps_type == "pip":
         requirements_files = []
-        if launch_project.project_dir is not None and os.path.exists(
-            os.path.join(launch_project.project_dir, "requirements.txt")
-        ):
+        deps_install_line = None
+        assert launch_project.project_dir is not None
+        base_path = pathlib.Path(launch_project.project_dir)
+        # If there is a requirements.txt at root of build context, use that.
+        if (base_path / "requirements.txt").exists():
             requirements_files += ["src/requirements.txt"]
-            pip_install_line = "pip install -r requirements.txt"
-        elif launch_project.project_dir is not None and os.path.exists(
-            os.path.join(launch_project.project_dir, "requirements.frozen.txt")
-        ):
-            # if we have frozen requirements stored, copy those over and have them take precedence
-            requirements_files += ["src/requirements.frozen.txt", "_wandb_bootstrap.py"]
-            pip_install_line = (
+            deps_install_line = "pip install -r requirements.txt"
+        # Elif there is pyproject.toml at build context, convert the dependencies
+        # section to a requirements.txt and use that.
+        elif (base_path / "pyproject.toml").exists():
+            tomli = get_module("tomli")
```

Comment:
nit: rather than this get_module + if, get_module allows for a second arg to raise a value error. Why not use that?

Reply:
I don't think I want to error if tomli isn't installed. Just warn that we couldn't use their pyproject.toml because `tomli` isn't installed, because we can still fall back on the `requirements.frozen.txt`. If that isn't present we raise a `LaunchError` below. Whatcha think?